### ISSUE TITLE
Broken Road Generation

### ISSUE THREAD

**Issue by knoxhack** [X-Niter/DTM-Integrations#2]
did you guys changed the world generation? because the road seem to be acting weird,

Mod Version: DTMIntegrations-1.12.2-1.2.0

[Screenshot](https://imgur.com/lx2pHpZ)

[Screenshot](https://imgur.com/IMSJaiZ)

**Comment by knoxhack** [comment]
yeah the road world gen is broken

[Imgur](https://imgur.com/MSl0MVA)

**Comment by X-Niter** [comment]
@knoxhack Thanks for letting me know, I tested this on a few SP maps and didn't come across this before updating. 
I did not change any math or generation behind the roads other than adding a safer check to make sure it doesn't try to replace a "Other mods" building with a road going right through it as it was very silly looking. 

That should not have manipulated any functionality behind the road generating, but, the original mod's code is a bit back and forth and a tad bit inter-looping making it easy to get lost in a wall of code blocks with no simple direction. 

I'll take a look here into what's going on with generation, I know one thing we're trying to tackle safely is making an attempt to clean up the generation as right now it's horrendous in that it cascades horribly.

There were also thoughts on completely recoding the mod and stripping away all the fancy GUI and rendering changes and making it more of a drop-in playable thing, but I still felt that this mod needs to at least make some attempt to clean things up for players who enjoy it on 1.12.2 at its natural state.
